content/js2/blocks/multiple-params/index.md

```diff
@@ -74,3 +78,28 @@ function parseQueryString(queryString) {
 {{</tab>}}
 
 {{</tabs>}}
+
+Now that we've worked out how to solve this problem in the case of multiple query parameters, let's integrate that solution into our previous implementation, to make sure it works for all cases.
+
+We can keep our `if (queryString.length === 0) {` check from before. We don't need to do anything special for the one-value case, as an array containing one element gets iterated the same as an array multiple elements:
+
+```js
+function parseQueryString(queryString) {
+  const queryParams = {};
+  if (queryString.length === 0) {
+    return queryParams;
+  }
+  const keyValuePairs = queryString.split("&");
+
+  for (const pair of keyValuePairs) {
+    const [key, value] = pair.split("=");
+    queryParams[key] = value;
+  }
+
+  return queryParams;
+}
+```
+
+When we're solving problems involving several values, often we need slightly differently handling for the cases when there are 0, 1, or more than 1 values. In our example here, we need to treat 0 values specially (if the query string is empty, we return early), but we can handle 1 and more than 1 the same way.
+
+When you're breaking down problems, think to yourself: What are special cases we may need to handle differently?
```

Comment:
@Dedekind561

Can we also talk about edge-cases here? Three important ones to identify are missing =s (`foo&bar=baz`), repeated keys (`foo=bar&foo=baz`), and empty segments (`?&foo=bar`). I think we should probably have a section on identifying the edge-cases, then set an exercise to work out how to handle them, and implement that?

Reply:
I'm afraid of there being too much content in the prep. Can we move all the edge case material into a separate exercise in the coursework? Or even do a workshop to explore edge cases and how to handle them?

Reply:
I'm happy to skip the exercise, but I'd like to keep the prompt to think about them... What do you think?

Reply:
Yes absolutely - I like the prompt